### PR TITLE
[Refactor] Move the logic for building the information_schema data in FrontendServiceImpl to InformationSchemaDataSource

### LINKED ISSUE #11221: Move the logic for building the information_schema data in FrontendServiceImpl to InformationSchemaDataSource
## Refactoring request

**Is your refactoring request related to a problem? Please describe.**

Move the logic for building the information_schema data in FrontendServiceImpl to InformationSchemaDataSource

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by waittttting**
…rviceImpl to InformationSchemaDataSource

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11221 

## Problem Summary(Required) ：

Move the logic for building the information_schema data in FrontendServiceImpl to InformationSchemaDataSource
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 81 / 85 (95.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|80|84|95.24%|[162, 163, 164, 167]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|1|1|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_clang

**Comment by waittttting** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_clang-tidy

**Comment by wangruin** [comment]
run starrocks_be_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wangruin** [comment]
run starrocks_fe_unittest

**Comment by wangruin** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
starrocks_clang-tidy

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_clang-tidy

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 84 (14.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|11|83|13.25%|[99, 100, 101, 103, 104, 105, 106, 107, 109, 110, 111, 116, 122, 125, 126, 128, 131, 143, 145, 146, 148, 149, 150, 151, 153, 154, 155, 157, 158, 159, 160, 161, 162, 163, 164, 165, 169, 170, 171, 172, 173, 175, 176, 178, 179, 180, 181, 182, 183, 185, 186, 187, 188, 189, 191, 192, 193, 195, 196, 197, 198, 202, 203, 205, 209, 210, 211, 212, 213, 214, 215, 216]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 80 / 84 (95.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|79|83|95.18%|[160, 161, 162, 165]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|1|1|100.00%|[]|

**Review by gengjun-git — DISMISSED** [review]

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -970,8 +970,16 @@ struct TSetConfigResponse {
...
+struct TAuthInfo {
+    // If not set, match every database
+    1: optional string pattern
+    2: optional string user   // deprecated
+    3: optional string user_ip    // deprecated
+    4: optional Types.TUserIdentity current_user_ident // to replace the user and user ip
+}
 
+struct TGetTablesConfigRequest {
+    1: required TAuthInfo auth_info
```
Comment (Astralidea) [inline gensrc/thrift/FrontendService.thrift:982]: This should be optional

Comment (waittttting) [reply gensrc/thrift/FrontendService.thrift]: done

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 80 / 84 (95.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|79|83|95.18%|[160, 161, 162, 165]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|1|1|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 80 / 84 (95.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|79|83|95.18%|[160, 161, 162, 165]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|1|1|100.00%|[]|

**Pull request merged**